### DISCUSSION TITLE
Sonoff SA-018 (No Neutral wire Switch)

### DISCUSSION THREAD

**Discussion by paulzhdanov** [RavenSystem/esp-homekit-devices#1980] [MEPLHAA Scripts Collection]
Hi guys,
I found cheap ($4.50) and good enough alternative to Aqara No Neutral Switch ($25), very useful thing if you wanna make smart light without crashing the walls and changing a hundred lamps. No idea if there's the way to flash it without soldering, but it's not a problem for me. 

GPIO0 - button
GPIO12 - relay
GPIO13 - blue led
GPIO4 - contacts (black wire) for physical switch

{
  "c":{"l":13,"b":[{"g":0,"t":5}],"n":"haalights"},
  "a":[{
    "0":{"r":[{"g":12}]},
    "1":{"r":[{"g":12,"v":1}]},
    "b":[{"g":0},{"g":4},{"g":4,"t":0}]
  }]
}

![IMG_7227](https://user-images.githubusercontent.com/13831213/223529948-b14263eb-a55d-42fa-9c03-ccc65b189d7a.jpeg)

**Comment by Dikvadik** [comment]
I did it to. It work's good

**Reply by paulzhdanov** [reply]
Did you flash it the same way? I know there is a DIY mode, but didn't succeed with it

**Comment by Dikvadik** [comment]
ср, 8 бер. 2023 р. о 17:55 paulzhdanov ***@***.***> пише:

> Did you flash it the same way? I know there is a DIY mode, but didn't
> succeed with it
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RavenSystem/esp-homekit-devices/discussions/1980#discussioncomment-5243419>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5QF7XLEWLJPSFMGJKOOE73W3CTX5ANCNFSM6AAAAAAVS4KEW4>
> .
> You are receiving this because you commented.Message ID:
> <RavenSystem/esp-homekit-devices/repo-discussions/1980/comments/5243419@
> github.com>
>
Yes, I do. I flash it the same way
